### PR TITLE
Macro items toggle

### PR DESCRIPTION
**Pull request by NejcZdovc**
This PR solves #122

### CODE REVIEW

**Comment by fjozsef** [comment]
LGTM, but I would want to wait [this anwser](https://github.com/UltimateHackingKeyboard/agent/issues/122#issuecomment-254598104), because the new macro action doesn't animate and I don't now if it should.

**Comment by fjozsef** [comment]
Okay, we need them.

**Comment by mondalaci** [comment]
Well, it's up to you guys whether to put the second animation into this PR or to another one.

**Comment by fjozsef** [comment]
It has definitely should be in this one.

**Comment by NejcZdovc** [comment]
It will be in this one

**Comment by NejcZdovc** [comment]
Added sliding effect for new item as well

**Comment by mondalaci** [comment]
This has made the user experience so much smoother!

The only problem I noticed is that now there's a 2 pixel wide separator line between macro actions versus the previous 1 pixel.

**Comment by NejcZdovc** [comment]
This was introduced in PR #87, but it's fixed now.

File: src/components/macro/macro.component.scss [hunk 1]
```diff
@@ -98,7 +101,7 @@ h1 {
 }
 
 .add-new__action-container {
-    border-top: 1px solid #ddd;
+    overflow: hidden;
```
Comment (fjozsef) [inline src/components/macro/macro.component.scss]: Could you also add `flex-shrink: 0;` ? 
When a macro is selected for edit the buttons at the bottom disappear for me, but it solves the problem.

File: src/components/macro/item/macro-item.component.ts [hunk 2]
```diff
@@ -25,6 +49,7 @@ export class MacroItemComponent implements OnInit, OnChanges {
     @Input() editable: boolean;
     @Input() deletable: boolean;
     @Input() moveable: boolean;
+    @Input() newItem: boolean;
```
Comment (fjozsef) [inline src/components/macro/item/macro-item.component.ts]: I think we shouldn't add one more input. What if we would assume its a new item if `macroAction` is `null`?

Comment (NejcZdovc) [inline src/components/macro/item/macro-item.component.ts]: Added.

**Pull request merged**